src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs

```diff
@@ -406,7 +406,7 @@ impl NonOverlapSubLevelPicker {
 
             let ret = self.pick_sub_level(l0, level_handler, sst);
             if ret.sstable_infos.len() < self.min_expected_level_count
-                && ret.total_file_size < self.min_compaction_bytes
+                || ret.total_file_size < self.min_compaction_bytes
```

Comment:
I guess the motivation to change `&&` to `||` here is that we choose to rely on L0 intra compaction to reduce the level count when `ret.sstable_infos.len() >= self.min_expected_level_count` so that we can ensure that to base compaction must contain at least `self.min_expected_level_count` and at least `self.min_compaction_bytes`. Let's document this in the codes.